src/app/modules/auth/__tests__/auth.service.spec.ts

```diff
@@ -187,4 +204,125 @@ describe('auth.service', () => {
       expect(actualResult._unsafeUnwrapErr()).toEqual(expectedError)
     })
   })
+
+  describe('getFormAfterPermissionChecks', () => {
+    const MOCK_USER = {
+      _id: new ObjectId(),
+    } as IPopulatedUser
+    it('should return form when user has permissions', async () => {
+      // Arrange
+      const mockFormId = new ObjectId().toHexString()
+      const expectedForm = {
+        title: 'mock form',
+        _id: mockFormId,
+      } as IPopulatedForm
+      MockFormService.retrieveFullFormById.mockReturnValueOnce(
+        okAsync(expectedForm),
+      )
+      const assertSpy = jest.fn().mockImplementation(() => () => ok(true))
```

Comment:
I don't think this needs to be so complicated hahahahaha
```
MockAdminFormUtils.getAssertPermissionFn.mockReturnValueOnce(() => ok(true))
const actualResult = ...
expet(MockAdminFormUtils.getAssertPermissionFn).toHaveBeenCalledWith(PermissionLevel.Write)
```

Reply:
I don't want to mock the entire util ma, but yea ok I'll see what I can do :P